libraries/freertos_plus/aws/greengrass/test/aws_test_greengrass_discovery.c

```diff
@@ -102,6 +109,28 @@ TEST_GROUP_RUNNER( Full_GGD )
     RUN_TEST_CASE( Full_GGD, GetGGCIPandCertificate );
 }
 
+static BaseType_t prvGGD_JSONRequestGetFileLoop( uint32_t ulBufferSize,
+                                                 uint32_t * pulByteRead,
+                                                 BaseType_t * pxJSONFileRetrieveCompleted,
+                                                 uint32_t ulJSONFileSize )
+{
+    BaseType_t xStatus = pdPASS;
+    /* Limit loop count to avoid any chance of infinite loops. */
+    uint32_t ulRequestLoopCounter = 0;
+
+    do
+    {
+        xStatus = GGD_JSONRequestGetFile( &xSocket,
+                                          &cBuffer[ *pulByteRead ],
+                                          ulBufferSize - *pulByteRead,
+                                          pulByteRead,
+                                          pxJSONFileRetrieveCompleted,
+                                          ulJSONFileSize );
+    } while( ( xStatus == pdPASS ) && ( *pxJSONFileRetrieveCompleted != pdTRUE ) && ( ulBufferSize > *pulByteRead ) && ( ++ulRequestLoopCounter < ggdTestMAX_REQUEST_LOOP_COUNT ) );
```

Comment:
Do we have to use ggdTestMAX_REQUEST_LOOP_COUNT ?  We are checking pxJSONFileRetrieveCompleted and ulBufferSize > pulByteRead, should these checks be sufficient to read the entire file and terminate if there is an error ?

Reply:
With our code, no, and that's why there's no loop counter for the similar code in the [library](https://github.com/aws/amazon-freertos/blob/master/libraries/freertos_plus/aws/greengrass/src/aws_greengrass_discovery.c#L193). However, since this is part of a test, I thought that we should guard against the possibility that a change to `GGD_JSONRequestGetFile` might break it in a way such that those variables don't get updated properly.